tests/core/policies/test_memoization.py

```diff
@@ -157,3 +227,306 @@ def test_augmented_prediction(self, max_history):
+                    type: bool
+            """
+        )
+        training_story = TrackerWithCachedStates.from_events(
+            "training story",
+            [
+                UserUttered(intent={"name": GREET_INTENT_NAME}),
+                ActionExecuted(UTTER_ACTION_1),
+                ActionExecuted(UTTER_ACTION_2),
+                ActionExecuted(UTTER_ACTION_3),
+                ActionExecuted(UTTER_ACTION_4),
+                ActionExecuted(UTTER_ACTION_5),
+                ActionExecuted(UTTER_BYE_ACTION),
+            ],
+            domain=domain,
+            slots=domain.slots,
+        )
+        interpreter = RegexInterpreter()
+        policy.train([training_story], domain, interpreter)
+
+        test_story1 = TrackerWithCachedStates.from_events(
+            "test story",
+            [
+                UserUttered(intent={"name": GOODBYE_INTENT_NAME}),
+                SlotSet("slot_1", False),
+                ActionExecuted(UTTER_BYE_ACTION),
+                UserUttered(intent={"name": GREET_INTENT_NAME}),
+                ActionExecuted(UTTER_ACTION_1),
+                ActionExecuted(UTTER_ACTION_2),
+                ActionExecuted(UTTER_ACTION_3),
+                ActionExecuted(UTTER_ACTION_4),
+                ActionExecuted(UTTER_ACTION_5),
+                # ActionExecuted(UTTER_BYE_ACTION),
+            ],
+            domain=domain,
+            slots=domain.slots,
+        )
+        prediction1 = policy.predict_action_probabilities(
+            test_story1, domain, interpreter
+        )
+        assert (
+            domain.action_names_or_texts[
+                prediction1.probabilities.index(max(prediction1.probabilities))
+            ]
+            == UTTER_BYE_ACTION
+        )
+
+        test_story2_no_match_expected = TrackerWithCachedStates.from_events(
+            "test story",
+            [
+                UserUttered(intent={"name": GREET_INTENT_NAME}),
+                SlotSet("slot_1", False),
+                ActionExecuted(UTTER_BYE_ACTION),
+                UserUttered(intent={"name": GOODBYE_INTENT_NAME}),
+                ActionExecuted(UTTER_ACTION_1),
+                ActionExecuted(UTTER_ACTION_2),
+                ActionExecuted(UTTER_ACTION_3),
+                ActionExecuted(UTTER_ACTION_4),
+                ActionExecuted(UTTER_ACTION_5),
+                # ActionExecuted(ACTION_LISTEN_NAME),
+            ],
+            domain=domain,
+            slots=domain.slots,
+        )
+
+        prediction2 = policy.predict_action_probabilities(
+            test_story2_no_match_expected, domain, interpreter
+        )
+        assert (
+            domain.action_names_or_texts[
+                prediction2.probabilities.index(max(prediction2.probabilities))
+            ]
+            == ACTION_LISTEN_NAME
+        )
+
+    @pytest.mark.parametrize("max_history", [1, 2, 3, 4, None])
+    def test_aug_pred_connects_different_memoizations(self, max_history):
```

Comment:
Great test case! Some clean-up could be done by removing the unused slots (`slot_2`, `slot_3`) and adding `action_listen` before user utterances.